### PR TITLE
feat(general): Repo key derivation algorithm

### PR DESCRIPTION
**Pull request by bathina2**
This PR adds support to set the key derivation algorithm when using repositories. It is a replacement for https://github.com/kopia/kopia/pull/3731

### CODE REVIEW

File: repo/format/content_format.go
Comment (jkowalski) [inline repo/format/content_format.go:21]: i don't understand how this works, `keyDerivationAlgorithm` must be known beforehand to generate key to decrypt this JSON, so by definition it can't be stored in this JSON, right?

Comment (bathina2) [reply repo/format/content_format.go:21]: When creating a new repository we pass this in as the NewRepositoryOptions. See `repo/initialize.go`.
Its also stored unencrypted in the status.

File: repo/initialize.go
Comment (jkowalski) [inline repo/initialize.go:112]: do we need to store it at all?

File: repo/format/content_format.go
Comment (jkowalski) [reply repo/format/content_format.go:21]: it should ONLY be stored unencrypted in `KopiaRepositoryJSON` and not in `format.ContentFormat` - the latter is encrypted using key derived using the algorithm itself, so it makes no sense.

Comment (jkowalski) [reply repo/format/content_format.go:21]: We should move `KeyDerivationAlgorithm` to be a field in `NewRepositoryOptions` instead of a field in `format.ContentFormat`

**Comment by bathina2** [comment]
Thanks for the comments and clarity @jkowalski. I addressed your feedback. Let me know if you want to see any other changes.

**Review by Shrekster — COMMENTED** [review]
Changes look good. @bathina2 can we put a test which verifies the connection using the key derivation algorithm flag from CLI. Also, one more test that verifies the behavior of repo connect when you change the algorithm,... would be useful to verify the error response when that happens.

**Review by Shrekster — APPROVED** [review]
LGTM TY for adding tests.

**Review by Shrekster — COMMENTED** [review]
Please wait for @julio-lopez 's approval before merge, he has some concerns to be addressed.

**Review by julio-lopez — CHANGES_REQUESTED** [review]
Please hold off merging

File: repo/format/repository_config.go
Comment (julio-lopez) [inline repo/format/repository_config.go]:
```suggestion
```

File: repo/initialize.go
Comment (julio-lopez) [inline repo/initialize.go]:
```suggestion
```

**Review by julio-lopez — COMMENTED** [review]
@bathina2 Please address the comments around:
- CLI flag name
- Field name in `NewRepositoryOptions struct`
- Remove unused `RepositoryConfig.KeyDerivationAlgorithm` field

**Review by julio-lopez — APPROVED** [review]

**Pull request merged**
